ext/mysql2/client.c

```diff
@@ -30,14 +30,21 @@ VALUE rb_hash_dup(VALUE other) {
     rb_raise(cMysql2Error, "MySQL client is not initialized"); \
   }
 
+#define REQUIRE_CLIENT(wrapper) \
```

Comment:
The naming of `GET_CLIENT` can be a bit confusing, since it actually gets the client wrapper.  `GET_CLIENT_WRAPPER` might be a more appropriate name.

Here are some functions that need the client wrapper without the `REQUIRE_CLIENT` check:
* **initialize_ext**: to set the client field
* **rb_mysql_client_close**: to be idempotent, not raising on the second close
* **rb_mysql_client_ping**: to return false when closed
* **rb_mysql_client_encoding**: value is in the wrapper
* **get_automatic_close**: value is in the wrapper
* **set_automatic_close**: value is in the wrapper